### PR TITLE
Remove constructors from fragments

### PR DESCRIPTION
**Pull request by nielsvanvelzen**
Depends on #669

**Changes**
- Removed constructor from NextUpFragment
  - Refactored the code to use a view model
- Remove constructor from StartupToolbarFragment
  - Simple approach by calling a function in the activity

**Todo**
- [ ] Remove constructor from AlertFragment
- [ ] Remove constructor from UserLoginFragment
- [ ] Remove constructor from AddServerFragment

**Issues**

### CODE REVIEW

**Comment by nielsvanvelzen** [comment]
I skipped the AlertFragment and subclasses because I expect it to be a big change, so it's easier to do in it's own PR.

File: app/src/main/java/org/jellyfin/androidtv/ui/playback/nextup/NextUpActivity.kt [hunk 1]
```diff
@@ -1,82 +1,44 @@
...
+			when (state) {
+				// Open next item
+				NextUpState.PLAY_NEXT -> {
+					startActivity(Intent(this, PlaybackOverlayActivity::class.java))
+					finish()
+				}
+				// Close activity
+				NextUpState.CLOSE -> finish()
+				// Unknown state
+				else -> Unit
```
Comment (thornbill) [inline app/src/main/java/org/jellyfin/androidtv/ui/playback/nextup/NextUpActivity.kt:27]: Should we log this?

Comment (nielsvanvelzen) [reply app/src/main/java/org/jellyfin/androidtv/ui/playback/nextup/NextUpActivity.kt:27]: The other possibilities are:
- NO_DATA, which is an indicator that we don't know the details of the next item, everything will work as normal but no episode title or image is shown
- INITIALIZED - Which is the default value and persists when data is correctly loaded

PLAY_NEXT and CLOSE are "actions" which this listener is meant to listen for, I therefore don't think logging is required.

**Review by thornbill — APPROVED** [review]

**Comment by thornbill** [comment]
Should we add an issue to track removing the other instances?

**Comment by nielsvanvelzen** [comment]
I'll probably get to those before 0.12 releases, we could probably add it to #612 though.

**Pull request merged**